server/interceptor/metrics.go

```diff
@@ -29,25 +31,28 @@ const (
 var (
 	requestCounter       monitoring.Counter
 	requestDeniedCounter monitoring.Counter
+
+	metricsOnce = sync.Once{}
 )
 
-// InitMetrics initializes the metrics on the interceptor package.
-func InitMetrics(mf monitoring.MetricFactory) {
-	requestCounter = mf.NewCounter("interceptor_request_count", "Total number of intercepted requests")
-	requestDeniedCounter = mf.NewCounter(
-		"interceptor_request_denied_count",
-		"Number of requests by denied, labeled according to the reason for denial",
-		"reason", monitoring.TreeIDLabel, "quota_user")
+func initMetrics(mf monitoring.MetricFactory) {
+	metricsOnce.Do(func() {
+		if mf == nil {
+			mf = monitoring.InertMetricFactory{}
+		}
+		quota.InitMetrics(mf)
+		requestCounter = mf.NewCounter("interceptor_request_count", "Total number of intercepted requests")
+		requestDeniedCounter = mf.NewCounter(
+			"interceptor_request_denied_count",
+			"Number of requests by denied, labeled according to the reason for denial",
+			"reason", monitoring.TreeIDLabel, "quota_user")
+	})
 }
 
 func incRequestCounter() {
```

Comment:
At this point, probably better just to drop the wrapper functions here and below.

(After that, it may also make sense to fold what's left of metrics.go into interceptor.go